aruco_ros/CMakeLists.txt

```diff
@@ -23,98 +21,112 @@ set(THIS_PACKAGE_INCLUDE_DEPENDS
     sensor_msgs
     visualization_msgs
 )
+
 find_package(ament_cmake REQUIRED)
-foreach(Dependency IN ITEMS ${THIS_PACKAGE_INCLUDE_DEPENDS})
-  find_package(${Dependency} REQUIRED)
+foreach(DEP ${dependencies})
+  find_package(${DEP} REQUIRED)
 endforeach()
 
-#generate_dynamic_reconfigure_options(
-#  cfg/ArucoThreshold.cfg
-#)
-
-add_library(aruco_ros_utils SHARED src/aruco_ros_utils.cpp)
-target_include_directories(aruco_ros_utils
+# Build library.
+add_library(${PROJECT_NAME}_utils SHARED
+  src/aruco_ros_utils.cpp
+)
+target_include_directories(${PROJECT_NAME}_utils
   PUBLIC
-  include)
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include/${PROJECT_NAME}>)
 
-target_include_directories(aruco_ros_utils
-  SYSTEM PUBLIC
+target_link_libraries(${PROJECT_NAME}_utils PUBLIC
+  ${OpenCV_LIBRARIES}
+  tf2_geometry_msgs::tf2_geometry_msgs
+  cv_bridge::cv_bridge
+)
+
+target_include_directories(${PROJECT_NAME}_utils PUBLIC
   ${OpenCV_INCLUDE_DIRS}
+  ${aruco_INCLUDE_DIRS}
+  ${sensor_msgs_INCLUDE_DIRS}
+  ${visualization_msgs_INCLUDE_DIRS}
 )
-ament_target_dependencies(aruco_ros_utils ${THIS_PACKAGE_INCLUDE_DEPENDS})
-target_link_libraries(aruco_ros_utils ${OpenCV_LIBRARIES})
 
+# Build executables.
 add_executable(single src/simple_single.cpp
                       src/aruco_ros_utils.cpp)
+target_link_libraries(single
+  ${PROJECT_NAME}_utils  
 
-target_include_directories(single
-  PUBLIC
-  include)
+  ${aruco_LIBRARIES}
+  ${OpenCV_LIBRARIES}
+  ${visualization_msgs_TARGETS}
 
-target_include_directories(single
-  SYSTEM PUBLIC
+  image_transport::image_transport
+)
+target_include_directories(single PUBLIC
   ${OpenCV_INCLUDE_DIRS}
 )
-ament_target_dependencies(single ${THIS_PACKAGE_INCLUDE_DEPENDS})
-target_link_libraries(single ${OpenCV_LIBRARIES})
 
 add_executable(double src/simple_double.cpp
                       src/aruco_ros_utils.cpp)
-target_include_directories(double
-  PUBLIC
-  include)
+target_link_libraries(double
+  ${PROJECT_NAME}_utils
+
+  ${aruco_LIBRARIES}
+  ${OpenCV_LIBRARIES}
 
-target_include_directories(double
-  SYSTEM PUBLIC
+  image_transport::image_transport
+)
+target_include_directories(double PUBLIC
   ${OpenCV_INCLUDE_DIRS}
 )
-ament_target_dependencies(double ${THIS_PACKAGE_INCLUDE_DEPENDS})
-target_link_libraries(double ${OpenCV_LIBRARIES})
 
 add_executable(marker_publisher src/marker_publish.cpp
                                 src/aruco_ros_utils.cpp)
-target_include_directories(marker_publisher
-  PUBLIC
-  include)
-
-target_include_directories(marker_publisher
-  SYSTEM PUBLIC
+target_link_libraries(marker_publisher
+  ${PROJECT_NAME}_utils
+  
+  ${aruco_LIBRARIES}
+  ${aruco_msgs_LIBRARIES}
+  ${OpenCV_LIBRARIES}
+  
+  image_transport::image_transport
```

Comment:
one could consider adding the `image_transport::image_transport` dependency to the `${PROJECT_NAME}_utils` to simplify the CMakeLists.txt